include/seastar/core/abort_source.hh

```diff
@@ -96,33 +103,45 @@ public:
 private:
     using subscription_list_type = bi::list<subscription, bi::constant_time_size<false>>;
     std::optional<subscription_list_type> _subscriptions = subscription_list_type();
+    std::exception_ptr _ex = nullptr;
 
 public:
     /// Delays the invocation of the callback \c f until \ref request_abort() is called.
     /// \returns an engaged \ref optimized_optional containing a \ref subscription that can be used to control
     ///          the lifetime of the callback \c f, if \ref abort_requested() is \c false. Otherwise,
     ///          returns a disengaged \ref optimized_optional.
+    template <typename Func>
     [[nodiscard]]
-    optimized_optional<subscription> subscribe(subscription_callback_type f) noexcept(std::is_nothrow_move_constructible<subscription_callback_type>::value) {
+    optimized_optional<subscription> subscribe(Func&& f) noexcept(std::is_nothrow_move_constructible_v<Func>) {
```

Comment:
Please add a SEASTAR_CONCEPT constraint to aid lucky C++20 users.

Reply:
Added in v2.